project/Mibudin/README.md

```diff
@@ -0,0 +1,251 @@
+
+Besides, this project has to provide friendly and widely flexible interfaces
+between the C++ library and Python codes, to make this project easy to be used
+in many kinds of applications.
+
+
+## Prospective Users
+
+The people who are interested or needed to use Wave Function Collapse as a
+procedural modeling method to produce large-scale textures or models that are
+local similar to given pixel/grid/tile-based examples. They may be artist,
+designers, game developers, etc.
+
+
+## API Description
+
+> [!WARNING]
+> The following API description only focuses on major exposed C++ API, which is
+> brief and may not be the final version.
+
+```cpp
+/**
+ * @brief The major WFC processor class
+ */
+class Model
+{
+public:
+    /**
+     * @brief Construct a new Model object
+     * 
+     * @param width the width of the result output
+     * @param height the height of the result output
+     * @param tiles the input tile-based example
+     * @param grammar the grammar of production rules
+     */
+    Model(size_t width, size_t height, TileSet tiles, Grammar grammar);
+
+    /**
+     * @brief Initialize this Model
+     */
+    void init();
+
+    /**
+     * @brief Run the whole WFC algorithm iteratively
+     * 
+     * @param seed the random seed of this processing
+     * @param limit the upper bound of the number of iterations
+     */
+    void run(int seed, int limit);
+
+    /**
+     * @brief Save the running result output to a file or a variable
+     * 
+     * @param path the path of the output file
+     */
+    void save(string path);
+};
+```
+
+
+## Engineering Infrastructure
+
+### C++-Python Interface
+
+- [pybind11][pybind11]
+
+### Automatic Build System
+
+- [GNU Make][gnu-make]
+- [CMake][cmake]
+
+### Version Control
+
+- [Git][git]
+- [GitHub][github]
+
+### Testing Framework
+
+- [pytest][pytest]
+
+### Continuous Integration
+
+- [GitHub Actions][gh-actions]
+
+### Documentation
+
+- [Markdown files][gfm]
+- [Doxygen-style documentation][doxygen-doc] (within C++ codes[^doc-comment])
+- [Google-style docstring][google-docstring] (within Python
+    codes[^doc-comment])
+
+
+## Schedule
+
+### Planning phase (6 weeks from 09/18 to 10/30)
+
+- Study and research various versions of implementations of Wave Function
+    Collapse, and select an appropriate one for this project
+- Try to plan a detailed API specification
+- Write the proposal
+
+### Week 1 (10/30)
+
+- Continuously try to plan a more concrete API specification
+- Plan a concrete architecture of the codes about the whole processing
+- Start to write the documents while developing
+
+### Week 2 (11/06)
+
+- Start to implement `Model`
+
+### Week 3 (11/13)
+
+- Implement `Model()`, `init()`
+- Start to write testing code following the development
+
+### Week 4 (11/20)
+
+- Implement `run()`
+
+### Week 5 (11/27)
+
+- Implement `save()`
+
+### Week 6 (12/04)
+
+- Implement the interface to Python using `pybind11` 
```

Comment:
If the binding is made so late, you won't have time to write any tests in Python.  There is not much time left for planning so you may need to adjust it when you go.  Please find me to discuss more frequently on github and discord.